### PR TITLE
Uvflag ant to baseline

### LINKED ISSUE #565: Let uvflag convert from antenna to baseline types

### LINKED ISSUE #652: Allow initialization of empty UVFlag object
Currently, UVFlag is initialized from a UVFlag file or existing UVData or UVCal objects. It would be nice to be able to initialize an empty UVFlag object, as we do for UVData and the other objects.

### LINKED ISSUE #683: remove weights_array for flag type UVFlag object?

### PR DESCRIPTION
**Pull request by mkolopanis**
Some updates to missing UVFlag features
## Description

Added the ability to initialize blank UVFlag objects. Can now covert from antenna type to baseline type objects.
## Motivation and Context

Closes #652  
Closes #565 
Closes #683 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

New feature checklist:
- [ ] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate).
- [x] I have added tests to cover my new feature.
- [x] All new and existing tests pass in both python 2 and python 3.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by mkolopanis** [comment]
The failures on this PR are still the miriad issues we've been seeing lately, other than that, I think it is ready for a review.

File: pyuvdata/uvflag.py [hunk 1]
```diff
@@ -19,6 +19,11 @@
 from . import utils as uvutils
 from . import telescopes as uvtel
 
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
```
Comment (adampbeardsley) [inline pyuvdata/uvflag.py]: get rid of

File: pyuvdata/uvflag.py [hunk 2]
```diff
@@ -1798,49 +1970,96 @@ def to_baseline(self, uv, force_pol=False, run_check=True,
...
+                self.metric_array = arr
+                self.weights_array = warr
+
+        elif self.type == "antenna":
+            if self.mode == "metric":
+                raise NotImplementedError("Cannot currently convert from "
+                                          "antenna type, metric mode to "
+                                          "baseline type UVFlag object.")
+            else:
+                if uv.Nants_data > self.Nants_data:
```
Comment (adampbeardsley) [inline pyuvdata/uvflag.py]: Is it reasonable to make this a little smarter? In principle you could have `uv.Nants_data < self.Nants_data`, but not all antennas in `uv` are in `self`.
In fact, I think you just need to run the `unique` and `setdiff1d` statements below and if `new_ants` has more than zero antennas, you do the rest.

File: pyuvdata/uvflag.py [hunk 3]
```diff
@@ -1798,49 +1970,96 @@ def to_baseline(self, uv, force_pol=False, run_check=True,
...
+                                                new_flags,
+                                                axis=0)
+
+                baseline_flags = np.full((uv.Nblts, self.Nspws,
+                                          self.Nfreqs, self.Npols),
+                                         True, dtype=bool)
+
+                for t_index, bl in enumerate(uv.baseline_array):
+                    # TODO: match times within the time_range if the UVFlag
+                    # object was created from a UVCal object
```
Comment (adampbeardsley) [inline pyuvdata/uvflag.py]: Should there be a `NotImplementedError` if this is the case? Or was this meant to be addressed before the PR?

File: pyuvdata/uvflag.py (hunk 2 above)
Comment (mkolopanis) [reply pyuvdata/uvflag.py]: yeah that makes a lot more sense.

File: pyuvdata/uvflag.py (hunk 3 above)
Comment (mkolopanis) [reply pyuvdata/uvflag.py]: Ah I have been thinking about this a little bit. I feel like we may actually want this to be smarter in the `from_uvcal` method. But there was some discussion and the consensus seemed to be that there would be very few use cases where the times would not match exactly anyway.

I'm willing to make a new issue to explore this use case as a needs research topic though.

**Pull request merged**